core/services/ocr2/plugins/functions/reporting.go

```diff
@@ -193,13 +194,17 @@ func (r *functionsReporting) Observation(ctx context.Context, ts types.ReportTim
 		// NOTE: ignoring TIMED_OUT requests, which potentially had ready results
 		if localResult.State == functions.RESULT_READY {
 			resultProto := encoding.ProcessedRequest{
-				RequestID: localResult.RequestID[:],
-				Result:    localResult.Result,
-				Error:     localResult.Error,
+				RequestID:       localResult.RequestID[:],
+				Result:          localResult.Result,
+				Error:           localResult.Error,
+				OnchainMetadata: localResult.OnchainMetadata,
 			}
 			if localResult.CallbackGasLimit != nil {
 				resultProto.CallbackGasLimit = *localResult.CallbackGasLimit
 			}
+			if localResult.CoordinatorContractAddress != nil {
```

Comment:
If this is nil, don't we need to have an error here?

Reply:
yes, for v1 we could error out, I'll add it